pkg/proxy/proxier.go

```diff
@@ -665,6 +725,18 @@ func iptablesInit(ipt iptables.Interface) error {
 	if _, err := ipt.EnsureRule(iptables.TableNAT, iptables.ChainOutput, "-j", string(iptablesHostPortalChain)); err != nil {
 		return err
 	}
+	if _, err := ipt.EnsureChain(iptables.TableNAT, iptablesContainerPublicChain); err != nil {
+		return err
+	}
+	if _, err := ipt.EnsureRule(iptables.TableNAT, iptables.ChainPrerouting, "-j", string(iptablesContainerPublicChain)); err != nil {
+		return err
+	}
+	if _, err := ipt.EnsureChain(iptables.TableNAT, iptablesHostPublicChain); err != nil {
+		return err
+	}
+	if _, err := ipt.EnsureRule(iptables.TableNAT, iptables.ChainOutput, "-j", string(iptablesHostPublicChain)); err != nil {
```

Comment:
I wrote that and then I had another thought - why do we need iptables here at all?

Shouldn't this be as simple as passing publicPort to addServiceOnPort() instead of 0?